### PR TITLE
Special Holiday Puzzles

### PR DESCRIPTION
**Pull request by canadianveggie**
Add explicit overwrites for Holidays that work sustainably into the future.

These were all done manually before with shuffling of words.

Useful for reviewing:
`node scripts/calendar.js`
`node scripts/holidays.js`

### CODE REVIEW

File: types.d.ts [hunk 1]
```diff
@@ -64,6 +64,19 @@ export type EmojiMatchThemes = {
   [key in EmojiMatchThemeKey]: EmojiMatchThemeValue;
 };
 
+export type DailyPuzzles = {
```
Comment (canadianveggie) [inline types.d.ts:67]: How do I test the typing again? Not sure I defined all these properly.

Comment (wachunga) [reply types.d.ts:68]: VS Code should do it inline automatically, but I'll add a script. We have a few issues

File: public/puzzles.js [hunk 2]
```diff
@@ -2459,44 +2385,276 @@ const start = new Date(2022, 2 - 1, 10);
 
 const MILLIS_PER_DAY = 1000 * 60 * 60 * 24;
 
+/**
+ * @param {Date} first
+ * @param {Date} second
+ * @returns int
```
Comment (wachunga) [inline public/puzzles.js:2391]: these should be `number`, I guess - but let's let the script catch it 😄

File: types.d.ts (hunk 1 above)
Comment (canadianveggie) [reply types.d.ts:68]: Cool. I loaded it in VS and fixed some of the type problems

File: public/puzzles.js (hunk 2 above)
Comment (canadianveggie) [reply public/puzzles.js:2391]: I think they are dates actually. I noticed they had problems when I loaded it into VS Code.

You can do `new Date() - new Date()` in javascript, but the type compiler doesn't like it.

Changed these to `new Date().getTime() - new Date().getTime()` which apparently is what the original gets coerced into.

Comment (wachunga) [reply public/puzzles.js:2391]: Oh, I just meant the type you're specifying, but I see now it's not a type (`{int}`) so I guess it's ok. Typescript only has `{number}`

**Comment by wachunga** [comment]
@canadianveggie here's the type checking script: https://github.com/canadianveggie/spellie/pull/136

I did it from the main branch to establish a baseline, but we can merge them in whichever order you prefer. There'll be at least 1 conflict, but hopefully not much more than that.

**Review by wachunga — APPROVED** [review]
Holidays LGTM. Nice word choices 💯

File: public/puzzles.js (hunk 2 above)
Comment (canadianveggie) [reply public/puzzles.js:2391]: oops!

**Comment by canadianveggie** [comment]
The conflicts resolved easily. Thanks for adding the type checking.

**Pull request merged**